### PR TITLE
Pratik/group chat

### PR DESCRIPTION
**Pull request by C0SMlC**

### CODE REVIEW

File: src/controllers/privateMessage.controller/getLatestPrivateMessage/index.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 const { validationResult } = require('express-validator');
 const { getPrivateMessages } = require('./queries/queries');
-const { decryptMessages } = require('../../../utils/decryptMessages');
+const { decryptMessage } = require('../../../utils/decryptMessages');
```
Comment (Shivz3232) [inline src/controllers/privateMessage.controller/getLatestPrivateMessage/index.js]: Please update the file name too

File: src/controllers/privateMessage.controller/getLatestPrivateMessage/index.js [hunk 2]
```diff
@@ -23,8 +23,23 @@ const getLatestPrivateMessage = catchAsync(async (req, res) => {
   };
 
   const response1 = await Hasura(getPrivateMessages, variables);
+  const messageDocs = response1.result.data.private_messages;
 
-  const decryptedMessages = await decryptMessages(response1.result.data.private_messages);
+  let decryptedMessages = [];
+
+  for (let i = 0; i < messageDocs.length; i++) {
```
Comment (Shivz3232) [inline src/controllers/privateMessage.controller/getLatestPrivateMessage/index.js]: Please use iterators like map etc. Its never good to have for loops in production code, only use if necessary.

File: src/controllers/teamMessage.controller/getMessage/index.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+const getMessage = catchAsync(async (req, res) => {
+  const sanitizerErrors = validationResult(req);
+  if (!sanitizerErrors.isEmpty()) {
+    return res.status(400).json({
+      success: false,
+      ...sanitizerErrors,
+    });
+  }
+  const { cognito_sub } = req.body;
```
Comment (Shivz3232) [inline src/controllers/teamMessage.controller/getMessage/index.js:16]: Body isn't available in a GET request

File: src/controllers/teamMessage.controller/getMessage/index.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+  // const response1 = await Hasura(getUserIdFromCognitoSub, { cognito_sub });
+  // const user_id = response1.result.data.user[0].id;
+
+  const variables = { team_id, cognito_sub };
+
+  const response1 = await Hasura(checkIfUserInTeam, variables);
+
+  if (response1.result.data.members.length === 0) {
+    return res.status(400).json(errorResponse('Forbidden', 'Given user is not a member of this team.'));
```
Comment (Shivz3232) [inline src/controllers/teamMessage.controller/getMessage/index.js:25]: errorResponse method not found

File: src/controllers/teamMessage.controller/getMessage/index.js [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+  const sanitizerErrors = validationResult(req);
+  if (!sanitizerErrors.isEmpty()) {
+    return res.status(400).json({
+      success: false,
+      ...sanitizerErrors,
+    });
+  }
+  const { cognito_sub } = req.body;
+  const { team_id, timeStamp } = req.query;
+  const limit = req.query.limit || 10;
```
Comment (Shivz3232) [inline src/controllers/teamMessage.controller/getMessage/index.js:18]: Derive limit too from req.query

File: src/controllers/teamMessage.controller/getMessage/index.js [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+  const response1 = await Hasura(checkIfUserInTeam, variables);
+
+  if (response1.result.data.members.length === 0) {
+    return res.status(400).json(errorResponse('Forbidden', 'Given user is not a member of this team.'));
+  }
+
+  const variables2 = {
+    team_id,
+    timeStamp: timeStamp || new Date().toISOString(),
+    limit,
```
Comment (Shivz3232) [inline src/controllers/teamMessage.controller/getMessage/index.js:31]: limit: limit || 10

File: src/controllers/teamMessage.controller/getMessage/index.js [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+  };
+
+  const response2 = await Hasura(getTeamMessages, variables2);
+  const messageDocs = response2.result.data.team_messages;
+
+  const decryptedMessages = [];
+
+  for (let i = 0; i < messageDocs.length; i++) {
```
Comment (Shivz3232) [inline src/controllers/teamMessage.controller/getMessage/index.js]: Please switch to iterator like map or each. For loop with i=0 is very bad in production code. Use for/while loops only when necessary.

File: src/controllers/teamMessage.controller/getMessage/queries/queries.js [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+
+const checkIfUserInTeam = `query checkIfUserInTeam($team_id: Int, $cognito_sub: String) {
+  members: team_members(where: {team_id: {_eq: $team_id}, user: {cognito_sub: {_eq: $cognito_sub}}}) {
+    user_id
+    admin
+    team {
+      creator_id
+    }
+  }
+  admins: team_members(where: {team_id: {_eq: $team_id}, user: {cognito_sub: {_eq: $cognito_sub}}, admin: {_eq: true}}) {
```
Comment (Shivz3232) [inline src/controllers/teamMessage.controller/getMessage/queries/queries.js]: Dont need admins here

File: src/controllers/privateMessage.controller/getLatestPrivateMessage/index.js (hunk 1 above)
Comment (C0SMlC) [reply src/controllers/privateMessage.controller/getLatestPrivateMessage/index.js]: Done

File: src/controllers/teamMessage.controller/getMessage/index.js (hunk 5 above)
Comment (C0SMlC) [reply src/controllers/teamMessage.controller/getMessage/index.js:18]: already doing that

**Pull request merged**